### PR TITLE
Add voluntary self identification form to applications and other partial S21 updates

### PR DESCRIPTION
**Pull request by sherryhli**
Partial updates for S21 applications, see commits for details.

**Voluntary self ID form**: I took some shortcuts here in the interest of getting this out quickly, please review accordingly.
* Assumption: a `voluntarySelfIdQuestion` is either a dropdown question, or a checkbox question
* Checkboxes can only be used in the context of the voluntary ID section

**Demo**:
demo

### CODE REVIEW

File: src/configs/application_config.ts [hunk 1]
```diff
@@ -224,6 +233,18 @@ export const APPLICATION_CONFIG: AppConfig = {
         { value: '3 times or more' }
       ]
     },
+    {
+        question: 'What is your pronoun?',
```
Comment (MickeyDang) [inline src/configs/application_config.ts:237]: Nit: I feel like "What is your preferred pronoun?" is more commonly used.

File: src/app/student-application/student-application.component.html [hunk 2]
```diff
@@ -182,6 +196,51 @@ <h2 class="uppercase">Role Specific Questions</h2>
                       studentApplicationForm.controls['roleQuestion' + (i + 1)].touched">Invalid response.</small>
       </div>
     </div>
+    <div class="row">
+        <h2 class="uppercase">Voluntary Self Identification Form</h2>
+        <p>
+            This section is completely optional. This information will be greatly valuable for improving diversity and inclusion efforts
```
Comment (MickeyDang) [inline src/app/student-application/student-application.component.html:202]: 👏 . Way more concise than the forms I see elsewhere lol.

**Review by MickeyDang — APPROVED** [review]
lol don't know angular well enough to thoroughly review, but i looked over the copy and the functions make sense to me fwiw.

File: src/configs/application_config.ts (hunk 1 above)
Comment (sarafong) [reply src/configs/application_config.ts:237]: yeah or "What are your preferred pronouns" is probably the way to go

**Comment by sarafong** [comment]
same as mickey, it looks pretty non offensive so LGTM 😄

**Review by sarafong — APPROVED** [review]

**Pull request merged**